dev/tasks/verify-rc/github.nix.yml

```diff
@@ -0,0 +1,70 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# NOTE: must set "Crossbow" as name to have the badge links working in the
+# github comment reports!
+name: Crossbow
+
+on:
+  push:
+    branches:
+      - "*-github-*"
+
+jobs:
+  verify:
+    name: "Verify release candidate {{ os }} {{ artifact }} {{ flag }}"
+    runs-on: {{ os }}-latest
+    steps:
+      - name: Checkout Arrow
+        run: |
+          git clone --no-checkout {{ arrow.remote }} arrow
+          git -C arrow fetch -t {{ arrow.remote }} {{ arrow.branch }}
+          git -C arrow checkout FETCH_HEAD
+          git -C arrow submodule update --init --recursive
+      - name: Fetch Submodules and Tags
+        shell: bash
+        run: cd arrow && ci/scripts/util_checkout.sh
+      - name: Run verification
+        shell: bash
+        run: |
+          set -e
+
+          {{ flag }}
+          if [ $(uname) = "Darwin" ]; then
+            brew install llvm@7 thrift glib
+          else
+            sudo apt-get --purge remove -y llvm-8 llvm-9
+            sudo apt-get install -y \
+              llvm-7 \
+              wget curl libboost-all-dev jq \
+              autoconf-archive gtk-doc-tools libgirepository1.0-dev flex bison
```

Comment:
I guess this will have to be revised for 0.17.0. I'm slightly perturbed that having multiple versions of LLVM causes a problem. Perhaps we could set the LLVM root directory (e.g. /usr/lib/llvm-$VERSION) instead?

Reply:
Yes, we'll have to revisit for 0.17 (though since llvm-8 is already on the ubuntu system, we shouldn't need to install anything).

FWIW removing the newer llvm versions didn't seem to solve the gandiva build problem, but maybe there was more to remove. Agreed that we shouldn't fail on it though, and @fsaintjacques had an idea for fixing the cmake, which we should try (just can't try in the context of 0.16 release artifacts).

Reply:
You also need to remove clang-8 and clang-9, this is where the mismatch is, precompiled headers are generated with clang-9 and llvm-links-7 is used to combine them in a single object file.